src/absil/ilwrite.fs

```diff
@@ -3875,7 +3874,28 @@ let writeDirectory os dict =
 
 let writeBytes (os: BinaryWriter) (chunk:byte[]) = os.Write(chunk,0,chunk.Length)  
 
-let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer: ILStrongNameSigner option, fixupOverlappingSequencePoints, emitTailcalls, showTimes, dumpDebugInfo) modul noDebugData =
+type EmitStreamProvider = Lazy<Choice<EmitTo, Diagnostic>>
+and EmitTo =
+    | EmittedFile of string
+    | EmittedStream of Stream
+and Diagnostic = string
+
+
+// Either
+let (|Success|Failure|) = function
```

Comment:
About that, i think `Success`/`Failure` make the code easier to read instead of `Choice1Of2` / `Choice2Of2` .

i can declare a type

```fsharp
type Result<'S, 'F> = Success of 'S | Failure of 'F
```

but Choice is the same, with a little help of active pattern. But maybe the custom DU is better?

I really like to refactor to change the code from exception workflow -> Success/Failure -> computation expression because is times easier to read and is easier to use